python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -685,13 +685,24 @@ def run(
         column_dist, column_dist_err = 'match_distance', 'match_distanceErr'
         dist = np.full(n_target, np.nan)
 
-        dist[matched_row] = np.hypot(
-            target.coord1[matched_row] - ref.coord1[matched_ref],
-            target.coord2[matched_row] - ref.coord2[matched_ref],
-        )
+        target_match_c1, target_match_c2 = (coord[matched_row] for coord in (target.coord1, target.coord2))
+        target_ref_c1, target_ref_c2 = (coord[matched_ref] for coord in (ref.coord1, ref.coord2))
+
         dist_err = np.full(n_target, np.nan)
-        dist_err[matched_row] = np.hypot(cat_target.iloc[matched_row][coord1_target_err].values,
-                                         cat_target.iloc[matched_row][coord2_target_err].values)
+        dist[matched_row] = sphdist(
+            target_match_c1, target_match_c2, target_ref_c1, target_ref_c2
+        ) if config.coord_format.coords_spherical else np.hypot(
+            target_match_c1 - target_ref_c1, target_match_c2 - target_ref_c2,
+        )
+        # Should probably explicitly add cosine terms if ref has errors too
```

Comment:
Are we ignoring the errors in the ref catalog because they're so tiny they don't really matter, or is it just overkill? Or are you thinking we should start paying attention to them sometime soon?

Reply:
Right now this is only used for DC2 and source injection, where the reference catalogs are truth with no errors. It will take a lot more work on a separate ticket to incorporate reference catalog errors.